### PR TITLE
Replace uWrite impl for String by generic impl for all Write impls

### PR DESCRIPTION
**Pull request by therealprof**
Signed-off-by: Daniel Egger <daniel@eggers-club.de>

### CODE REVIEW

File: write/src/lib.rs [hunk 1]
```diff
@@ -37,12 +36,12 @@ pub trait uWrite {
...
 
-#[cfg(feature = "std")]
-impl uWrite for String {
+impl<T: core::fmt::Write> uWrite for T
+{
     type Error = Infallible;
 
     fn write_str(&mut self, s: &str) -> Result<(), Infallible> {
-        self.push_str(s);
+        self.write_str(s).ok();
```
Comment (bjorn3) [inline write/src/lib.rs]: This should not silently ignore the error.

Comment (therealprof) [reply write/src/lib.rs]: Why not? That's what the default impl for `String` did, too. Kinds of makes sense to me.

What alternatives would you have in mind?

Comment (bjorn3) [reply write/src/lib.rs]: `type Error = core::fmt::Error;`

Comment (therealprof) [reply write/src/lib.rs]: Fair enough. I've put the alternative implementation into https://github.com/japaric/ufmt/pull/19 .

I'm expecting that one to potentially create overhead the avoidance of which is the whole point of this crate.

**Comment by japaric** [comment]
Closing in favor of #21

**Pull request closed without merging**